extensions/shared/block-keywords.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { _x } from '@wordpress/i18n';
+import { includes } from 'lodash';
+
+/**
+ * Filters block registration to add keywords to core blocks.
+ *
+ * @param {Object} settings Block Settings.
+ * @param {string} name     Block name.
+ * @return {Object} Filtered block settings.
+ */
+function addBlockKeywords( settings, name ) {
+	const social = _x( 'social', 'block search term', 'jetpack' );
+	const survey = _x( 'survey', 'block search term', 'jetpack' );
+	const video = _x( 'video', 'block search term', 'jetpack' );
+
+	switch ( name ) {
+		case 'core-embed/facebook':
+		case 'core-embed/instagram':
+		case 'core-embed/tumblr':
+		case 'core-embed/twitter':
+			if ( ! includes( settings.keywords, social ) ) {
+				settings.keywords.push( social );
+			}
+			break;
+
+		case 'core-embed/crowdsignal':
+			if ( ! includes( settings.keywords, survey ) ) {
+				settings.keywords.push( survey );
+			}
+			break;
+
+		case 'core-embed/hulu':
+			if ( ! includes( settings.keywords, video ) ) {
+				settings.keywords.push( video );
+			}
+			break;
```

Comment:
Should we open a core editor issue to add these keywords? They make more sense to me there but it's nice we can test them sooner with this 👍

Reply:
This definitely feels like a better fit for Core overall. We can add this in Jetpack for now, but it would be nice if we had a clear path to deprecating this (e.g. a matching Gutenberg PR and a comment here inviting the removal of the filter once we only support WP 5.4+, for example).

Reply:
I added a file comment with a reference